### PR TITLE
Add is_vendor attribute to Namespace

### LINKED ISSUE #555: As a system admin, I want to indicate that a Namespace represents a partner, so that we can highlight vendor-produced content
In the UI, we want to differentiate community contributed content from vendor contributed content. To support this, we need to know which Namespaces represent a partner. 

- [ ] Add 'partner' boolean attribute to Namespace
- [ ] Surface 'partner' in the Namspace serializer. Check Content, Repository, Provider_Namespaces, and surface 'partner' where Namespace is part of Summary Fields
- [ ] Default to False for all existing Namespaces

### PR DESCRIPTION
**Pull request by alikins**
Adds Namespace 'vendor' boolean #555

Default to False.

Include the field in NamespaceSerializer fields.

Add the 'partner' field to summary_fields['namespace']
for Repository, ProviderNamespace. Content
uses local _NamespaceSerializer so add the field to
cls.fields there.

Add 0098_namespace_partner.py migration

Fixes #555

### CODE REVIEW

**Comment by tima** [comment]
Perhaps we should use the term "vendor" (any company or organization even) here so we have the option of using "partner" (there is a more formal relationship established between Red Hat Ansible and the vendor) later.

File: galaxy/api/views/namespace.py [hunk 1]
```diff
@@ -258,7 +258,7 @@ def update(self, request, *args, **kwargs):
                 owners.append(request.user.pk)
             update_owners(instance, owners)
 
-        for item in ('name', 'description', 'avatar_url', 'location', 'company', 'email', 'html_url', 'active'):
+        for item in ('name', 'description', 'avatar_url', 'location', 'company', 'email', 'html_url', 'active', 'partner'):
```
Comment (chouseknecht) [inline galaxy/api/views/namespace.py]: I think 'partner' needs to be added above, on line 215, in the 'post' method of NamespaceList. It wouldn't let me add a comment to that line directly, but take a look.

**Comment by chouseknecht** [comment]
From a code and functionality standpoint this looks good to me. One small comment in the namespace view.

Sent @tima a follow-up email asking for resolution on 'partner' vs 'vendor'. Good question, and something we should definitely resolve now, if possible.

**Comment by chouseknecht** [comment]
@alikins 

From @tima, please change `partner` to `vendor`.

File: galaxy/api/views/namespace.py (hunk 1 above)
Comment (alikins) [reply galaxy/api/views/namespace.py]: Added.

**Comment by alikins** [comment]
rebased and amended to squash migrations to single 0098_namespace_vendor.py script.

File: galaxy/main/models.py [hunk 2]
```diff
@@ -540,6 +540,8 @@ class Meta:
         verbose_name="Web Site URL"
     )
 
+    vendor = models.BooleanField(default=False)
```
Comment (cutwater) [inline galaxy/main/models.py]: Can you please name it `is_vendor`

File: galaxy/api/views/namespace.py [hunk 3]
```diff
@@ -258,7 +258,7 @@ def update(self, request, *args, **kwargs):
                 owners.append(request.user.pk)
             update_owners(instance, owners)
 
-        for item in ('name', 'description', 'avatar_url', 'location', 'company', 'email', 'html_url', 'active'):
+        for item in ('name', 'description', 'avatar_url', 'location', 'company', 'email', 'html_url', 'active', 'vendor'):
```
Comment (cutwater) [inline galaxy/api/views/namespace.py]: Probably we shouldn't update `vendor` field from API. It is only to be managed by admin users.

Comment (chouseknecht) [reply galaxy/api/views/namespace.py]: By 'admin' we mean user.is_staff=True, not the Django admin. I think what @alikins has is correct. 

There is a separate issue, #433, that covers limiting Namespace create/delete to users where user.is_staff. We'll add the permission checks as part of that issue.

Comment (cutwater) [reply galaxy/api/views/namespace.py]: Thanks for the clarification.

**Comment by chouseknecht** [comment]
Changed 'vendor' to 'is_vendor'.

**Pull request merged**